### ISSUE TITLE
Wrong "no-value-for-parameter" warnings when using a metaclass

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#205]
Originally reported by: **Damien Baty (BitBucket: [dbaty](http://bitbucket.org/dbaty), GitHub: @dbaty?)**

---

Using `factory-boy` (https://pypi.python.org/pypi/factory_boy), I get wrong warnings when using a subclass of a `DjangoModelFactory`, like this:

```
import factory

class MyFactory(factory.DjangoModelFactory):
    """My factory."""

    value = 1

    def a_method(self, to_please_pylint):
        """Avoid 'too-few-public-methods' warning."""
        pass


INST = MyFactory(value=2)
```

Pylint warns about wrong parameters:

```
E: 19, 7: Unexpected keyword argument 'value' in constructor call (unexpected-keyword-arg)
E: 19, 7: No value for argument 'class_name' in constructor call (no-value-for-parameter)
E: 19, 7: No value for argument 'bases' in constructor call (no-value-for-parameter)
E: 19, 7: No value for argument 'attrs' in constructor call (no-value-for-parameter)
```

Perhaps Pylint is missing the fact that factory_boy uses a metaclass. See 
[here](https://github.com/rbarrois/factory_boy/blob/master/factory/django.py#L55) for the definition of `DjangoModelFactory`, and [there](https://github.com/rbarrois/factory_boy/blob/master/factory/base.py#L628) for the definition of its base class, `Factory`.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/205
